core/renderers/measurables/types.js

```diff
@@ -0,0 +1,353 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2019 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Measurable types.
+ * @author samelh@google.com (Sam El-Husseini)
+ */
+
+'use strict';
+
+goog.provide('Blockly.blockRendering.Types');
```

Comment:
Some thoughts
- It would be nice if we could find a way to make this name shorter, since we're using it explicitly everywhere.
- If this is a namespace, `types` should not be capitalized.

Reply:
Types is a singleton class, similar to ``Blockly.utils.IdGenerator``, so I think capitalized makes sense. Open to ideas on shortening the name :)

Reply:
Okay.  maybe we'll find a better name than Blockly.blockRendering for the first half.